src/common/KokkosKernels_SparseUtils.hpp

```diff
@@ -830,6 +728,177 @@ inline size_t kk_is_d1_coloring_valid(
   return num_conf;
 }
 
+template<typename execution_space, typename rowmap_t, typename entries_t, typename values_t>
+struct SortCrsMatrixFunctor
+{
+  using size_type = typename rowmap_t::non_const_value_type;
+  using lno_t = typename entries_t::non_const_value_type;
+  using scalar_t = typename values_t::non_const_value_type;
+  using team_mem = typename Kokkos::TeamPolicy<execution_space>::member_type;
+
+  SortCrsMatrixFunctor(bool usingRangePol, const rowmap_t& rowmap_, const entries_t& entries_, const values_t& values_)
+    : rowmap(rowmap_), entries(entries_), values(values_)
+  {
+    if(usingRangePol)
+    {
+      entriesAux = entries_t(Kokkos::ViewAllocateWithoutInitializing("Entries aux"),
+          entries.extent(0));
+      valuesAux = values_t(Kokkos::ViewAllocateWithoutInitializing("Values aux"),
+          values.extent(0));
+    }
+    //otherwise, aux arrays won't be allocated (sorting in place)
+  }
+
+  KOKKOS_INLINE_FUNCTION void operator()(const lno_t i) const
+  {
+    size_type rowStart = rowmap(i);
+    size_type rowEnd = rowmap(i + 1);
+    lno_t rowNum = rowEnd - rowStart;
+    //Radix sort requires unsigned keys for comparison
+    using unsigned_lno_t = typename std::make_unsigned<lno_t>::type;
+    KokkosKernels::Impl::SerialRadixSort2<lno_t, unsigned_lno_t, scalar_t>(
+        (unsigned_lno_t*) entries.data() + rowStart,
+        (unsigned_lno_t*) entriesAux.data() + rowStart,
+        values.data() + rowStart,
+        valuesAux.data() + rowStart, rowNum);
+  }
+
+  KOKKOS_INLINE_FUNCTION void operator()(const team_mem t) const
+  {
+    size_type i = t.league_rank();
```

Comment:
Are we assuming here that the `league_size` is equal to the number of rows to sort? That seems a bit risky, if the user does not realize that they might end up with a half-sorted `CrsMatrix`... Would it be OK to create the team policy inside the functor to make sure if verify this hypothesis.